File: Section Reports/Layout/SubReport/C#/SubReport/rptUnboundDSMain.cs
using System;
using System.Data;
namespace GrapeCity.ActiveReports.Samples.SubReport
{
	/// <summary>
	/// Description of the outline of the rptUnboundDSMain.
	/// 
	/// </summary>
	public partial class rptUnboundDSMain : GrapeCity.ActiveReports.SectionReport
	{
		rptUnboundDSSub _rpt;
		public rptUnboundDSMain()
		{
			//
			// ActiveReport designer support is required.
			//
			//
			InitializeComponent();
		}
		private void Detail_Format(object sender, EventArgs e)
		{
			// Create a DataView from the Orders table in the DataSet.
			//
			DataSet ds = new DataSet();
			ds = (DataSet)DataSource;
			DataView dv = new DataView(ds.Tables["Orders"]);
			dv.RowFilter = "CustomerID='" + TextBox1.Text + "'";
			// Orders data to bind to the report.
			//
		
			_rpt.DataSource = dv;
		   
			// Show the Orders report in the SubReport1 control.
			//
			SubReport1.Report = _rpt;
		}
		private void rptUnboundDSMain_ReportStart(object sender, EventArgs e)
		{
			_rpt = new rptUnboundDSSub();
			
			reportInfo1.FormatString = "Page {PageNumber} of {PageCount}";
			
			
		}
	}
}

File: Section Reports/Layout/SubReport/VB.NET/SubReport/rptUnboundDSMain.vb
Public Class rptUnboundDSMain
	Inherits GrapeCity.ActiveReports.SectionReport
	Dim _rpt As rptUnboundDSSub
	Public Sub New()
		MyBase.New()
		InitializeComponent()
	End Sub
	Private Sub Detail_Format(ByVal sender As Object, ByVal e As EventArgs) Handles Detail.Format
		'Create a DataView from the Orders table in the DataSet.
		'
		Dim ds As New DataSet()
		ds = CType(DataSource, DataSet)
		Dim dv As New DataView(ds.Tables("Orders"))
		dv.RowFilter = "CustomerID='" + TextBox1.Text + "'"
		'Orders data to bind to the report.
		'
		_rpt.DataSource = dv
		'Show the Orders report in the SubReport1 control.
		'
		SubReport1.Report = _rpt
	End Sub
	Private Sub rptUnboundDSMain_ReportStart(ByVal sender As System.Object, ByVal e As EventArgs) Handles MyBase.ReportStart
		_rpt = New rptUnboundDSSub()
		
		ReportInfo1.FormatString = "Page {PageNumber} of {PageCount}"
		
		'
	End Sub
End Class
